pkg/build/interfaces.go

```diff
@@ -49,6 +49,12 @@ type Ignorer interface {
 	Ignore(*api.Config) error
 }
 
+// Runner allows the user to actually run the produced imgae as part of the s2i
+// cmd line invocation, to see if it operates as he would expect
+type Runner interface {
+	Run(*api.Config) error
+}
```

Comment:
Don't see why being used inside the build or related to the build matters wrt leveraging an interface.  I see the interfaces as a way to outline and define the semantics of major portions of the entire program - concepts that are of interest to the user.